src/go/debug-collector/archive/archive.go

```diff
@@ -0,0 +1,58 @@
+package archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func Create(src string) error {
+	var buf bytes.Buffer
+
+	zr := gzip.NewWriter(&buf)
+	tw := tar.NewWriter(zr)
+
+	filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
+		header, err := tar.FileInfoHeader(fi, file)
+		if err != nil {
+			return err
```

Comment:
add more context to errors. given the app is going be run by users on their systems, if something goes wrong we want to have as much context as possible

Reply:
https://github.com/percona/percona-toolkit/pull/452/commits/a65dbefbd9ac88dfbeda1f3d8cd704049139170d